modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -4629,3 +4636,179 @@ def from_dataframe(cls, df: "ProtocolDataframe") -> "PandasDataframe":
         ErrorMessage.default_to_pandas(message="`from_dataframe`")
         pandas_df = from_dataframe_to_pandas(df)
         return cls.from_pandas(pandas_df)
+
+    def case_when(self, caselist):
+        """
+        Replace values where the conditions are True.
+
+        Parameters
+        ----------
+        caselist : list of tuples
+
+        Returns
+        -------
+        PandasDataframe
+        """
+        # For Dask the callables must be wrapped for each partition, otherwise
+        # the execution could fail with CancelledError.
+        single_wrap = Engine.get() != "Dask"
+        cls = type(self)
+        wrapper_put = self._partition_mgr_cls._execution_wrapper.put
+        if (
+            not single_wrap
+            or (remote_fn := getattr(cls, "_CASE_WHEN_FN", None)) is None
+        ):
+
+            def case_when(df, name, caselist):  # pragma: no cover
+                caselist = [
+                    tuple(
+                        (
+                            data.squeeze(axis=1)
+                            if isinstance(data, pandas.DataFrame)
+                            else data
+                        )
+                        for data in case_tuple
+                    )
+                    for case_tuple in caselist
+                ]
+                return pandas.DataFrame({name: df.squeeze(axis=1).case_when(caselist)})
+
+            if single_wrap:
+                cls._CASE_WHEN_FN = remote_fn = wrapper_put(case_when)
+            else:
+                remote_fn = case_when
+
+        name = self.columns[0]
+        use_map = single_wrap
+        is_trivial_idx = None
+        # Lists of modin frames: first for conditions, second for replacements
+        modin_lists = [[], []]
+        # Fill values for conditions and replacements respectively
+        fill_values = [True, None]
+        new_caselist = []
+        for case_tuple in caselist:
+            new_case = []
+            for data, modin_list, fill_value in zip(
+                case_tuple, modin_lists, fill_values
+            ):
+                if isinstance(data, cls):
+                    modin_list.append(data)
+                elif callable(data):
+                    if single_wrap:
+                        data = wrapper_put(data)
+                elif isinstance(data, pandas.Series):
+                    use_map = False
+                    if is_trivial_idx is None:
+                        self_idx = self.index
+                        length = len(self_idx)
+                        is_trivial_idx = is_trivial_index(self_idx)
+                    if is_trivial_idx and is_trivial_index(data.index):
+                        data = data[:length]
+                        diff = length - len(data)
+                        if diff > 0:
+                            data = pandas.concat(
+                                [data, pandas.Series([fill_value] * diff)]
```

Comment:
```suggestion
                                [data, pandas.Series([fill_value] * diff)], ignore_index=True
```
we probably need to pass `ignore_index=True` here, otherwise the output index would be incorrect:
```python
>>> data
0    1
1    2
2    3
3    4
dtype: int64
>>> pandas.concat([data, pandas.Series([0] * 4)])
0    1
1    2
2    3
3    4
0    0
1    0
2    0
3    0
dtype: int64
```